### PR TITLE
Fixes for Windows

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
This PR contains a minimal set of changes from https://github.com/ros-infrastructure/rosdep/pull/661 to get the RoboStack installer (https://github.com/ros-infrastructure/rosdep/pull/810) to work on Windows.

@nuclearsandwich already had a look at https://github.com/ros-infrastructure/rosdep/pull/661 and the last comment was to rework the Azure pipeline to use GitHub actions instead; in this PR I'm taking a more minimal approach and only pull in easy-to-review changes without touching tests etc. which we don't need for our RoboStack use-case.

The contents of the PR are already used within the conda-forge build of rosdep (see https://github.com/conda-forge/rosdep-feedstock). We confirmed that Windows works in our moveit PR to add cross-platform CI (https://github.com/ros-planning/moveit/pull/2604) which makes use of rosdep.

/cc @wolfv @traversaro

### CODE REVIEW

File: src/rosdep2/main.py [hunk 1]
```diff
@@ -153,7 +153,10 @@ def rosdep_main(args=None):
     except UsageError as e:
         print(_usage, file=sys.stderr)
         print('ERROR: %s' % (str(e)), file=sys.stderr)
-        sys.exit(os.EX_USAGE)
+        if hasattr(os, 'EX_USAGE'):
+            sys.exit(os.EX_USAGE)
+        else:
+            sys.exit()
```
Comment (nuclearsandwich) [inline src/rosdep2/main.py:159]: Should this be a non-zero exit code to indicate an error even on platforms where EX_USAGE is not defined?

**Review by nuclearsandwich — COMMENTED** [review]
One comment about the change in exit code on error. If I was certain the exit code should be non-zero I would make the change myself and merge the PR but I want to get feedback from @Tobias-Fischer to make sure there's not some reason the program should exit cleanly here.

File: src/rosdep2/main.py (hunk 1 above)
Comment (Tobias-Fischer) [reply src/rosdep2/main.py:159]: Yes, I believe so. @traversaro - you are the Windows guru, is there a standard way of doing so or is simply "1" sufficient?

Comment (Tobias-Fischer) [reply src/rosdep2/main.py]: `os.EX_USAGE` returns 64 on Unix, so I decided to return the same non-zero value on Windows.

**Comment by Tobias-Fischer** [comment]
All done now - would be great if you could have another look @nuclearsandwich

**Review by nuclearsandwich — APPROVED** [review]

**Comment by nuclearsandwich** [comment]
Thanks @Tobias-Fischer!

**Pull request merged**